### PR TITLE
feat: add error popup during failed or long loading screens

### PR DESCRIPTION
**Pull request by Kinerius**
## What does this PR change?

Adds an error popup when one of these 2 conditions are met:
- ~~10 seconds passed after any`WebRequest` fails.~~ (removed due to a suggestion)
- 100 seconds passed since the loading of the scene started.

This popup gives 2 options, either to `Exit` or to `Reload`

Fixes https://github.com/decentraland/unity-renderer/issues/1492

## How to test the changes?

Its hard to test these changes so you will have to hack `WebRequestController` by changing their code:

Add a random fail chance after the web request is sent:
```c#
UnityWebRequestAsyncOperation requestOp = resultOp.webRequest.SendWebRequest();
            
if (Random.Range(0, 10) > 8)
{
      OnFail?.Invoke(resultOp);
      resultOp.SetAsCompleted(false);
      OnWebRequestFailed?.Invoke(url);
      return resultOp;
}
````

Run the game and test the popup.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by kuruk-mm — CHANGES_REQUESTED** [review]

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Utils/DesktopUtils.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    public static class DesktopUtils
+    {
+        public static void Quit()
+        {
+#if UNITY_EDITOR
+            // Application.Quit() does not work in the editor so
+            // UnityEditor.EditorApplication.isPlaying need to be set to false to end the game
+            UnityEditor.EditorApplication.isPlaying = false;
+#else
+            Application.Quit();
```
[the remark this answers is not in the document]
Comment (Kinerius) [reply unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Utils/DesktopUtils.cs:14]: Done

**Review by kuruk-mm — APPROVED** [review]

File: unity-renderer-desktop/Packages/manifest.json [hunk 2]
```diff
@@ -1 +1 @@
-{ "dependencies": { "com.coffee.git-dependency-resolver": "https://github.com/mob-sakai/GitDependencyResolverForUnity.git", "com.decentraland.unity-renderer": "git+https://github.com/decentraland/unity-renderer.git?path=unity-renderer/Assets#master", "com.unity.cinemachine": "2.7.4", "com.unity.testtools.codecoverage": "1.1.0", "com.unity.toolchain.linux-x86_64": "0.1.18-preview", "net.tnrd.nsubstitute": "https://github.com/decentraland/Unity3D-NSubstitute.git" }, "testables": [ "com.decentraland.unity-renderer" ] }
+{ "dependencies": { "com.coffee.git-dependency-resolver": "https://github.com/mob-sakai/GitDependencyResolverForUnity.git", "com.decentraland.unity-renderer": "git+https://github.com/decentraland/unity-renderer.git?path=unity-renderer/Assets#feat/loading-error", "com.unity.cinemachine": "2.7.4", "com.unity.testtools.codecoverage": "1.1.0", "com.unity.toolchain.linux-x86_64": "0.1.18-preview", "net.tnrd.nsubstitute": "https://github.com/decentraland/Unity3D-NSubstitute.git" }, "testables": [ "com.decentraland.unity-renderer" ] }
```
Comment (kuruk-mm) [inline unity-renderer-desktop/Packages/manifest.json:1]: update version to master before merging

**Review by kuruk-mm — APPROVED** [review]

**Review by RominaMarchetti — APPROVED** [review]

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        public void Update()
+        {
+            if (isDisposed) return;
+
+            if (Time.unscaledTime - timerStart > GENERAL_TIMEOUT_IN_SECONDS)
+            {
+                view.ShowForTimeout();
+                Dispose();
```
Comment (kuruk-mm) [inline unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs:44]: Do we really want to `Dispose()` the controller here?

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        private readonly ILoadingFlowView view;
+        private bool isDisposed;
+
+        public LoadingFlowController(Action reloadAction)
+        {
+            CommonScriptableObjects.rendererState.OnChange += OnRendererStateChange;
+            
+            view = CreateView();
+            view.Setup(reloadAction);
+            view.Hide();
```
Comment (kuruk-mm) [inline unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs:21]: We're hiding the view `gameObject.SetActive(false);` in the constructor and after we never do an `gameObject.SetActive(true);`

**Review by kuruk-mm — CHANGES_REQUESTED** [review]

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs (hunk 4 above)
Comment (Kinerius) [reply unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs:21]: Solved

File: unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs (hunk 3 above)
Comment (Kinerius) [reply unity-renderer-desktop/Assets/Scripts/MainScripts/DCL/Controllers/LoadingFlow/LoadingFlowController.cs:44]: What do you suggest? the only job of this class is to show the error on timeout and then stop working

**Pull request merged**